source/windows/iocp/iocp_event_loop.c

```diff
@@ -496,7 +496,13 @@ static int s_connect_to_io_completion_port(struct aws_event_loop *event_loop, st
         0,                   /* CompletionKey */
         1);                  /* NumberOfConcurrentThreads */
 
-    if (!success) {
+    /*
+     * Some handles, notably named pipes, don't like to be associated more than
+     * once with the same I/O completion port. Therefore, it's ok if this call
+     * fails as long as there's already a bound iocp_handle.
+     * This will result in a last error of ERROR_INVALID_PARAMETER.
+     */
+    if (!success && impl->iocp_handle == INVALID_HANDLE_VALUE) {
```

Comment:
Something is weird here.
`impl` is the `iocp_loop`. I don't know how `impl->iocp_handle` could possibly be `INVALID_HANDLE_VALUE`.

Here are the docs for [CreateIoCompletionPort](https://docs.microsoft.com/en-us/windows/win32/fileio/createiocompletionport#return-value). Could we call `GetLastError()` to determine the specific edge-case we're trying to allow here?

Reply:
It's ```INVALID_HANDLE_VALUE``` when it goes in for some cases (I believe for normal sockets). It comes out with a valid handle to the IOCP. If the IOCP is invalid, then this is a failure.